ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoTransactionBuilder.kt

```diff
@@ -0,0 +1,228 @@
+     * @return Returns a [UtxoTransactionBuilder] including the specified time window.
+     */
+    fun setTimeWindowFrom(from: Instant): UtxoTransactionBuilder
+
+    /**
+     * Sets the transaction time window to be valid until the specified [Instant], tending towards negative infinity.
+     *
+     * @param until The [Instant] until which the transaction time window is valid.
+     * @return Returns a [UtxoTransactionBuilder] including the specified time window.
+     */
+    fun setTimeWindowUntil(until: Instant): UtxoTransactionBuilder
+
+    /**
+     * Sets the transaction time window to be valid between the specified [Instant] values.
+     *
+     * @param from The [Instant] from which the transaction time window is valid.
+     * @param until The [Instant] until which the transaction time window is valid.
+     * @return Returns a [UtxoTransactionBuilder] including the specified time window.
+     */
+    fun setTimeWindowBetween(from: Instant, until: Instant): UtxoTransactionBuilder
+
+    /**
+     * Sets the transaction time window to be valid at the specified [Instant] with the specified tolerance.
+     *
+     * @param midpoint The [Instant] representing the midpoint of the time window.
+     * @param tolerance The [Duration] representing the tolerance of the time window, which is applied before and after the midpoint.
+     * @return Returns a [UtxoTransactionBuilder] including the specified time window.
+     */
+    fun setTimeWindowBetween(midpoint: Instant, tolerance: Duration): UtxoTransactionBuilder
+
+    /**
+     * Signs the transaction with any required signatories that belong to the current node.
+     *
+     * @return Returns a [UtxoSignedTransaction] with signatures for any required signatories that belong to the current node.
+     */
+    fun sign(): UtxoSignedTransaction
+
+    /**
+     * Signs the transaction with the specified signatory keys.
+     *
+     * @param signatories The signatories expected to sign the current transaction.
+     * @return Returns a [UtxoSignedTransaction] with signatures for the specified signatory keys.
+     */
+    fun sign(signatories: Iterable<PublicKey>): UtxoSignedTransaction
+
+    /**
+     * Signs the transaction with the specified signatory keys.
+     *
+     * @param signatories The signatories expected to sign the current transaction.
+     * @return Returns a [UtxoSignedTransaction] with signatures for the specified signatory keys.
+     */
+    fun sign(vararg signatories: PublicKey): UtxoSignedTransaction
+
+    /**
+     * Verifies the current transaction.
+     */
+    fun verify()
+
+    /**
+     * Verifies and signs the transaction with any required signatories that belong to the current node.
+     *
+     * @return Returns a [UtxoSignedTransaction] with signatures for any required signatories that belong to the current node.
+     */
+    fun verifyAndSign(): UtxoSignedTransaction
```

Comment:
Do we logically need a verify and sign method or is that just a convenience shortcut?

Reply:
Convenient shortcut.